src/Runner.Worker/ExecutionContext.cs

```diff
@@ -248,9 +251,15 @@ public void ForceTaskComplete()
             });
         }
 
-        public void RegisterPostJobStep(string refName, IStep step)
+        public void RegisterPostJobStep(IStep step)
         {
-            step.ExecutionContext = Root.CreatePostChild(step.DisplayName, refName, IntraActionState);
+            if (step is IActionRunner actionRunner && !Root.StepsWithPostRegistered.Add(actionRunner.Action.Id))
+            {
+                Trace.Info($"'post' of '{actionRunner.DisplayName}' already push to post step stack.");
+                return;
+            }
```

Comment:
Why do we need to do the check? Do we need to worry about duplicate post steps?

Reply:
Yes.